typing/ctype.ml

```diff
@@ -1295,26 +1295,42 @@ let new_local_type ?(loc = Location.none) ?manifest_and_scope () =
     type_uid = Uid.mk ~current_unit:(Env.get_unit_name ());
   }
 
-let existential_name cstr ty =
-  match get_desc ty with
-  | Tvar (Some name) -> "$" ^ cstr.cstr_name ^ "_'" ^ name
-  | _ -> "$" ^ cstr.cstr_name
+let existential_name name_counter ty =
+  let name =
+    match get_desc ty with
+    | Tvar (Some name) -> name
+    | _ ->
+        let name =
+          if !name_counter < 26
+          then String.make 1 (Char.chr(97 + !name_counter))
+          else String.make 1 (Char.chr(97 + !name_counter mod 26))
+               ^ Int.to_string(!name_counter / 26)
+        in
```

Comment:
I grabbed this logic from [printtyp.ml](https://github.com/ocaml/ocaml/blob/3afd70a70b9f8b305eabd2b06be76997718b0cd6/typing/printtyp.ml#L972). Would you prefer I update that code too, abstract the logic out into `misc.ml` and update it, or leave that part alone?

Reply:
I would certainly prefer if we shared the code somewhere instead of duplicating it, and if the code was as nice as reasonably possible, so I vote Misc.

Note: Printtyp is careful to avoid conflicts in the environment -- there is a counter but also a separate freshness check. Do I correctly understand that we don't need to avoid conflicts with user names because existential names are a separate namespace in which users cannot create names? (This might be worth a small comment somewhere.)